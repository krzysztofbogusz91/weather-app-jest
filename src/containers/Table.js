import React from 'react';

export const Table = (props) => {
    const isToday = props.type === "today";
    const title = isToday ? 'CURRENT WEATHER' : `FORECAST FOR NEXT 5 DAYS`;

    const data = isToday ? "Name" : "Date";

    const displayList = [];

    props.table.forEach((a, i, arr) => {
        //get only date form string
        const name1 = a.name.substring(0, a.name.indexOf(' '));
        if (arr.length > 1) {
            if (i === 0) {
                //for first element push first date
                displayList.push({ name: name1 })
            }
            if (i < arr.length - 1) {

                const elem2 = arr[i + 1];
                const name2 = elem2.name.substring(0, elem2.name.indexOf(' '));

                //compare dates if diffrent push date tr    
                if (name1 === name2) {

                    displayList.push(a)

                } else {
                    displayList.push(a)
                    //for every other day push next date
                    displayList.push({ name: name2 })
                }
            }
        } else {
            //for the today list push first and only  element
            displayList.push(a)
        }
    })


    const list = displayList.map((a, i) => {
        //if name is converted to date only its length is always 10
        if (a.name.length === 10) {
<<<<<<< develop
            return <tr className="bg-primary text-light" key={i + a.name}><th colSpan="5">{a.name}</th></tr>
=======
            return <tr key={i + a.name}><th colSpan="5">{a.name}</th></tr>
>>>>>>> master
        }
        const name = isToday ? a.name : a.name.substring(a.name.indexOf(' '), a.name.lenght);

        return (
            <tr key={i + a.name}>
                <td>{name}</td>
                <td><img alt="icon" src={`http://openweathermap.org/img/w/${a.icon}.png`} /></td>
                <td>{a.description}</td>
                <td> {Math.round(a.temp - 272.15) + ' C'}</td>
                <td>{a.humidity + ' %'}  </td>
            </tr>
        )
    })
    
    return (
        <table className="table table-striped text-center table-bordered table-light mt-4">
            <thead>
                <tr>
                    <th className='title' scope="col" colSpan="5">{title}</th>
                </tr>
                <tr>
                    <th scope="col">{data}</th>
                    <th scope="col">Icon</th>
                    <th scope="col">Description</th>
                    <th scope="col">Temperature</th>
                    <th scope="col">Humidity</th>
                </tr>
            </thead>
            <tbody className='tbody-class'>
                {list}
            </tbody>
        </table>
    );
}




