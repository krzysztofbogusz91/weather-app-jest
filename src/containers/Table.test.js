import  React  from 'react';
import set from '../enzyme_set_up';
import  { Table }  from './Table';
import { shallow, mount } from 'enzyme';


describe('Table', () => {
    const props = {type: "today", table: []}
    props.table = [
        {
            description:"clear sky",
            humidity:45,
            icon:"01d",
            id:800,
            main:"Clear",
            name:"2018-05-07 18:00:00",
            temp:294.31
        },
        {
            description:"clear sky",
            humidity:45,
            icon:"01d",
            id:800,
            main:"Clear",
            name:"2018-05-07 18:00:00",
            temp:294.31
        },
    ]
    const table = shallow(<Table {...props} />)
    it('should render properly', () => {
        expect(table).toMatchSnapshot()
    });

    it('should display title current weatcher', () => {
        table.setProps({
            type: "today"
        })
        expect(table.find('.title').text()).toEqual('CURRENT WEATHER')
    });
    
    it('should display title current weatcher', () => {
        table.setProps({
            type: "forecast"
        })
        expect(table.find('.title').text()).toEqual('FORECAST FOR')
    });

    it('should have length of form rows as passed arr', () => {
        expect(table.find('.tbody-class').find('tr').length).toEqual(2)
    });
});
