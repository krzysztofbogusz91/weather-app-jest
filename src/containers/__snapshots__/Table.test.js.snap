// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    table={
        Array [
            Object {
              "description": "clear sky",
              "humidity": 45,
              "icon": "01d",
              "name": "2018-05-07 18:00:00",
              "temp": 294.31,
            },
            Object {
              "description": "clear sky",
              "humidity": 45,
              "icon": "01d",
              "name": "2018-05-07 18:00:00",
              "temp": 294.31,
            },
          ]
    }
    type="today"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
                    <th
                              className="title"
                              colSpan="5"
                              scope="col"
                    >
                              CURRENT WEATHER
                    </th>
          </tr>
          <tr>
                    <th
                              scope="col"
                    >
                              Name
                    </th>
                    <th
                              scope="col"
                    >
                              Icon
                    </th>
                    <th
                              scope="col"
                    >
                              Description
                    </th>
                    <th
                              scope="col"
                    >
                              Temperature
                    </th>
                    <th
                              scope="col"
                    >
                              Humidity
                    </th>
          </tr>
</thead>,
        <tbody
          className="tbody-class"
>
          <tr>
                    <th
                              colSpan="5"
                    >
                              2018-05-07
                    </th>
          </tr>
          <tr>
                    <td>
                              2018-05-07 18:00:00
                    </td>
                    <td>
                              <img
                                        alt="icon"
                                        src="http://openweathermap.org/img/w/01d.png"
                              />
                    </td>
                    <td>
                              clear sky
                    </td>
                    <td>
                               
                              22 C
                    </td>
                    <td>
                              45 %
                                
                    </td>
          </tr>
</tbody>,
      ],
      "className": "table table-striped text-center table-bordered table-light mt-4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <th
                            className="title"
                            colSpan="5"
                            scope="col"
              >
                            CURRENT WEATHER
              </th>
</tr>,
            <tr>
              <th
                            scope="col"
              >
                            Name
              </th>
              <th
                            scope="col"
              >
                            Icon
              </th>
              <th
                            scope="col"
              >
                            Description
              </th>
              <th
                            scope="col"
              >
                            Temperature
              </th>
              <th
                            scope="col"
              >
                            Humidity
              </th>
</tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <th
                className="title"
                colSpan="5"
                scope="col"
>
                CURRENT WEATHER
</th>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CURRENT WEATHER",
                "className": "title",
                "colSpan": "5",
                "scope": "col",
              },
              "ref": null,
              "rendered": "CURRENT WEATHER",
              "type": "th",
            },
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  scope="col"
>
                  Name
</th>,
                <th
                  scope="col"
>
                  Icon
</th>,
                <th
                  scope="col"
>
                  Description
</th>,
                <th
                  scope="col"
>
                  Temperature
</th>,
                <th
                  scope="col"
>
                  Humidity
</th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Icon",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "Icon",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "Description",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Temperature",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "Temperature",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Humidity",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "Humidity",
                "type": "th",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <th
                            colSpan="5"
              >
                            2018-05-07
              </th>
</tr>,
            <tr>
              <td>
                            2018-05-07 18:00:00
              </td>
              <td>
                            <img
                                          alt="icon"
                                          src="http://openweathermap.org/img/w/01d.png"
                            />
              </td>
              <td>
                            clear sky
              </td>
              <td>
                             
                            22 C
              </td>
              <td>
                            45 %
                              
              </td>
</tr>,
          ],
          "className": "tbody-class",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "02018-05-07",
            "nodeType": "host",
            "props": Object {
              "children": <th
                colSpan="5"
>
                2018-05-07
</th>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2018-05-07",
                "colSpan": "5",
              },
              "ref": null,
              "rendered": "2018-05-07",
              "type": "th",
            },
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": "12018-05-07 18:00:00",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                  2018-05-07 18:00:00
</td>,
                <td>
                  <img
                                    alt="icon"
                                    src="http://openweathermap.org/img/w/01d.png"
                  />
</td>,
                <td>
                  clear sky
</td>,
                <td>
                   
                  22 C
</td>,
                <td>
                  45 %
                    
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2018-05-07 18:00:00",
                },
                "ref": null,
                "rendered": "2018-05-07 18:00:00",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="icon"
                    src="http://openweathermap.org/img/w/01d.png"
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "icon",
                    "src": "http://openweathermap.org/img/w/01d.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "clear sky",
                },
                "ref": null,
                "rendered": "clear sky",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "22 C",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "22 C",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "45 %",
                    "  ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "45 %",
                  "  ",
                ],
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
                        <th
                                    className="title"
                                    colSpan="5"
                                    scope="col"
                        >
                                    CURRENT WEATHER
                        </th>
            </tr>
            <tr>
                        <th
                                    scope="col"
                        >
                                    Name
                        </th>
                        <th
                                    scope="col"
                        >
                                    Icon
                        </th>
                        <th
                                    scope="col"
                        >
                                    Description
                        </th>
                        <th
                                    scope="col"
                        >
                                    Temperature
                        </th>
                        <th
                                    scope="col"
                        >
                                    Humidity
                        </th>
            </tr>
</thead>,
          <tbody
            className="tbody-class"
>
            <tr>
                        <th
                                    colSpan="5"
                        >
                                    2018-05-07
                        </th>
            </tr>
            <tr>
                        <td>
                                    2018-05-07 18:00:00
                        </td>
                        <td>
                                    <img
                                                alt="icon"
                                                src="http://openweathermap.org/img/w/01d.png"
                                    />
                        </td>
                        <td>
                                    clear sky
                        </td>
                        <td>
                                     
                                    22 C
                        </td>
                        <td>
                                    45 %
                                      
                        </td>
            </tr>
</tbody>,
        ],
        "className": "table table-striped text-center table-bordered table-light mt-4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <th
                                className="title"
                                colSpan="5"
                                scope="col"
                >
                                CURRENT WEATHER
                </th>
</tr>,
              <tr>
                <th
                                scope="col"
                >
                                Name
                </th>
                <th
                                scope="col"
                >
                                Icon
                </th>
                <th
                                scope="col"
                >
                                Description
                </th>
                <th
                                scope="col"
                >
                                Temperature
                </th>
                <th
                                scope="col"
                >
                                Humidity
                </th>
</tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <th
                  className="title"
                  colSpan="5"
                  scope="col"
>
                  CURRENT WEATHER
</th>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CURRENT WEATHER",
                  "className": "title",
                  "colSpan": "5",
                  "scope": "col",
                },
                "ref": null,
                "rendered": "CURRENT WEATHER",
                "type": "th",
              },
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    scope="col"
>
                    Name
</th>,
                  <th
                    scope="col"
>
                    Icon
</th>,
                  <th
                    scope="col"
>
                    Description
</th>,
                  <th
                    scope="col"
>
                    Temperature
</th>,
                  <th
                    scope="col"
>
                    Humidity
</th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Icon",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "Icon",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "Description",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Temperature",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "Temperature",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Humidity",
                    "scope": "col",
                  },
                  "ref": null,
                  "rendered": "Humidity",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <th
                                colSpan="5"
                >
                                2018-05-07
                </th>
</tr>,
              <tr>
                <td>
                                2018-05-07 18:00:00
                </td>
                <td>
                                <img
                                                alt="icon"
                                                src="http://openweathermap.org/img/w/01d.png"
                                />
                </td>
                <td>
                                clear sky
                </td>
                <td>
                                 
                                22 C
                </td>
                <td>
                                45 %
                                  
                </td>
</tr>,
            ],
            "className": "tbody-class",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "02018-05-07",
              "nodeType": "host",
              "props": Object {
                "children": <th
                  colSpan="5"
>
                  2018-05-07
</th>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2018-05-07",
                  "colSpan": "5",
                },
                "ref": null,
                "rendered": "2018-05-07",
                "type": "th",
              },
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "12018-05-07 18:00:00",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    2018-05-07 18:00:00
</td>,
                  <td>
                    <img
                                        alt="icon"
                                        src="http://openweathermap.org/img/w/01d.png"
                    />
</td>,
                  <td>
                    clear sky
</td>,
                  <td>
                     
                    22 C
</td>,
                  <td>
                    45 %
                      
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2018-05-07 18:00:00",
                  },
                  "ref": null,
                  "rendered": "2018-05-07 18:00:00",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="icon"
                      src="http://openweathermap.org/img/w/01d.png"
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "icon",
                      "src": "http://openweathermap.org/img/w/01d.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "clear sky",
                  },
                  "ref": null,
                  "rendered": "clear sky",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "22 C",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "22 C",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "45 %",
                      "  ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "45 %",
                    "  ",
                  ],
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
